openlibrary/plugins/upstream/utils.py

```diff
@@ -682,6 +696,29 @@ def normalize(s: str) -> str:
             continue
 
 
+def get_abbrev_from_full_lang_name(input_lang_name: str) -> str:
+    """
+    Take a language name, in English, such as 'English' or 'French' and return
+    'eng' or 'fre', respectively.
+
+    If there's no match, this returns an empty string.
+    If there are multiple matches, raise a LanguageMatchError.
+    """
+    possible_lang_abbrevs = autocomplete_languages(input_lang_name)
+    matches = 0
+    target_abbrev = ""
+
+    for abbrev_lang in possible_lang_abbrevs:
```

Comment:
Might best to iterate over `get_languages()` directly instead of calling autocomplete_languages; it should just be like 3 nested loops and a yield 😋

Reply:
I handled this slightly differently, but I think with the same result. I can refactor if desired. :D